metaflow/runner/utils.py

```diff
@@ -74,6 +74,32 @@ def read_from_file_when_ready(
         return content
 
 
+async def async_read_from_file_when_ready(
+    file_path: str, command_obj: "CommandManager", timeout: float = 5
+):
+    await asyncio.wait_for(command_obj.process.wait(), timeout)
+
+    with open(file_path, "r", encoding="utf-8") as file_pointer:
+        content = file_pointer.read()
+        if not content:
+            raise CalledProcessError(
+                command_obj.process.returncode, command_obj.command
+            )
+        return content
+
+
+def handle_process_error(command_obj: "CommandManager"):
+    stdout_log = open(command_obj.log_files["stdout"]).read()
+    stderr_log = open(command_obj.log_files["stderr"]).read()
+    command = " ".join(command_obj.command)
+    error_message = "Error executing: '%s':\n" % command
+    if stdout_log.strip():
+        error_message += "\nStdout:\n%s\n" % stdout_log
+    if stderr_log.strip():
+        error_message += "\nStderr:\n%s\n" % stderr_log
+    raise RuntimeError(error_message)
```

Comment:
you can probably return the `error_message` here...

and use that to do `raise RuntimeError(error_message) from e` in other places..

Reply:
You're right, this was a little silly of me. Fixed.